ynr/apps/candidates/tests/test_api.py

```diff
@@ -253,249 +225,258 @@ def test_api_post(self):
-    def test_cache_api_to_directory_prune_four_old(self, mock_datetime, mock_rmtree):
-        # Need to make sure datetime.strptime still works:
-        mock_datetime.strptime.side_effect = datetime.strptime
-        mock_datetime.now.return_value = datetime(2017, 5, 14, 12, 33, 5, 0)
-        target_directory = mkdtemp()
-        try:
-            for d in (
-                    '2017-05-12T06:00:00',
-                    '2017-05-12T08:00:00',
-                    '2017-05-12T10:00:00',
-                    '2017-05-12T12:00:00',
-            ):
-                os.makedirs(join(target_directory, d))
-            prune(target_directory)
-            # Even though all of those directories are more than 36
-            # hours old, they should all be kept because they're the
-            # most recent 4:
-            self.assertEqual(mock_rmtree.call_count, 0)
-        finally:
-            rmtree(target_directory)
-
-    @patch('candidates.management.commands.candidates_cache_api_to_directory.rmtree')
-    @patch('candidates.management.commands.candidates_cache_api_to_directory.datetime')
-    def test_cache_api_to_directory_prune_lots_very_recent(self, mock_datetime, mock_rmtree):
+        target_directory = settings.MEDIA_ROOT
+        call_command(
+            "candidates_cache_api_to_directory",
+            page_size="3",
+            url_prefix="https://example.com/media/api-cache-for-wcivf/",
+        )
+        expected_leafname = "2017-05-14T12:33:05"
+        expected_timestamped_directory = join(
+            settings.MEDIA_ROOT, "cached-api", expected_leafname
+        )
+        expected_symlink = join(settings.MEDIA_ROOT, "cached-api", "latest")
+        self.assertTrue(exists(expected_timestamped_directory))
+        self.assertTrue(exists(expected_symlink))
+        # Check that the files in that directory are as expected:
+        entries = os.listdir(expected_timestamped_directory)
+        persons_1_leafname = "persons-000001.json"
+        persons_2_leafname = "persons-000002.json"
+        posts_1_leafname = "posts-000001.json"
+        posts_2_leafname = "posts-000002.json"
+        self.assertEqual(
+            set(entries),
+            {
+                persons_1_leafname,
+                persons_2_leafname,
+                posts_1_leafname,
+                posts_2_leafname,
+            },
+        )
+        # Get the data from those pages:
+        with open(
+            join(expected_timestamped_directory, persons_1_leafname)
+        ) as f:
+            persons_1_data = json.load(f)
+        with open(
+            join(expected_timestamped_directory, persons_2_leafname)
+        ) as f:
+            persons_2_data = json.load(f)
+        with open(join(expected_timestamped_directory, posts_1_leafname)) as f:
+            posts_1_data = json.load(f)
+        with open(join(expected_timestamped_directory, posts_2_leafname)) as f:
```

Comment:
I know we're running the unit tests against a `FileSystemStorage` backend so we can just use the standard library here, but is it really easy to change these tests to also use the `DefaultStorage` API? That would allow us to run the tests against another storage engine at some point if we wanted (and possibly pick up this sort of issue earlier). If its any more complicated than replacing these calls to `open()` with `self.storage.open()` its probably not worth bothering with right now.